packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2779,6 +2779,20 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       return super.parseMaybeDecoratorArguments(expr);
     }
 
+    checkCommaAfterRest(close) {
+      if (this.match(tt.comma)) {
+        if (this.lookaheadCharCode() === close) {
+          if (!this.state.isDeclareContext) {
+            this.raiseTrailingCommaAfterRest(this.state.start);
+            return;
+          }
+          this.eat(tt.comma);
+        } else {
+          this.raiseRestNotLast(this.state.start);
```

Comment:
It is need to add tests to check raising this error.

Reply:
This method is inherited from here adding some adjustments:
https://github.com/babel/babel/blob/7bc72bb451e71d7d8964efdf4438d5a907aae655/packages/babel-parser/src/parser/lval.js#L526-L535
That branch is already tested in "ES" parser, I don't know if it is necessary to repeat those.

Reply:
Then I'd prefer to do something like this rather than copy-pasting the code:

```js
checkCommaAfterRest(close) {
  if (
    this.state.isDeclareContext &&
    this.match(tt.comma) &&
    this.lookaheadCharCode() === close
  ) {
    this.next(); // tt.comma
  } else {
    super.checkCommaAfterRest(close);
  }
}
```

by doing so, if we'll make any change to `checkCommaAfterRest` we don't risk forgetting to also update this second implementation.

Reply:
Don't we risk to call twice `lookaheadCharCode` doing so?

Reply:
Just thinking, maybe we can add some parser contributing.md section (or somewhere) mentioning that for plugins (TS/Flow) we would prefer overwritten methods to use super? Not sure if that's worth making a lint thing either though

Reply:
> Not sure if that's worth making a lint thing either though

It's hard to do so, because we need to know which methods shadow methods on the superclass. Maybe it's doable with tslint, but not with eslint.

Reply:
It's doable with this https://devblogs.microsoft.com/typescript/announcing-typescript-4-3-beta/#override-and-the-noimplicitoverride-flag

Reply:
Ah yeah I was thinking of a simple heuristic of methods that don't start with `ts` since it seems we do that? Not a big deal